spec/features/courses/subjects/new_spec.rb

```diff
@@ -155,6 +155,25 @@
       it_behaves_like "a course creation page"
     end
 
+    context "Not selecting master subject" do
+      let(:course) do
+        c = build(:course, :new, provider: provider, level: level, gcse_subjects_required_using_level: true, edit_options: edit_options)
+        c.errors.add(:subjects, "Invalid")
+        c
```

Comment:
You can use the [`tap`](https://apidock.com/ruby/Object/tap) method to make this a bit more concise:

```suggestion
        build(:course, :new, provider: provider, level: level, gcse_subjects_required_using_level: true, edit_options: edit_options).
          tap { |course| course.errors.add(:subjects, "Invalid") }
```

Reply:
Thanks @benilovj , i've made those changes